Kudu.Core/Settings/DiagnosticsSettingsManager.cs

```diff
@@ -69,6 +69,12 @@ private DiagnosticsSettings ReadSettings()
                 {
                     string fileContent = null;
                     OperationManager.Attempt(() => fileContent = FileSystemHelpers.ReadAllTextFromFile(_path));
+                    if (DiagnosticsSettings.AutoCorrect(_tracer, fileContent, out fileContent))
+                    {
+                        // if there is corrupted value, reset it back to file system
+                        FileSystemHelpers.WriteAllTextToFile(_path, fileContent);
+                    }
+
                     return JsonConvert.DeserializeObject<DiagnosticsSettings>(fileContent);
```

Comment:
Try running `JsonConvert.DeserializeObject<DiagnosticsSettings>(fileContent); ` with corrupted fileContent.  It will throw, no?